### ISSUE TITLE
dbms.read_only = true causes neo4j v3.4.18 to become unresponsive and report "this IndexReader is closed" after org.apache.lucene.store.AlreadyClosedException exceptions

### ISSUE THREAD

**Issue by jhpoelen** [neo4j/neo4j#12773]
First thanks for helping to build, use, and maintain neo4j . I've been gratefully using it for years now, and have been impressed by the care with which the neo4j codebase has been maintained and improved.

- Neo4j version: 3.4.18
- Operating system: Ubuntu 18.04 LTS
- API/Driver: errors are reported via http cypher endpoints as well as bolt /cypher-shell 

## steps to reproduce:
1. install neo4j v3.4.18 via apt-get 
2. create graph.db with legacy/explicit lucene indexes
3. set dbms.read_only = true in /etc/neo4j/neo4j.conf
## Expected behavior
Neo4j runs normally and any write operation fails. 

## Actual behavior
Neo4j becomes unresponsive after reporting a org.apache.lucene.store.AlreadyClosedException exceptions with recurring error message ```this IndexReader is closed```. 

## Work around

from /etc/neo4j/neo4j.conf -

```
...
# Only allow read operations from this Neo4j instance. This mode still requires
# write access to the directory for lock purposes.
#dbms.read_only=true
#
# only after disabling read-only mode in neo4j v3.4.18,
# periodic (~ every 10 minutes) org.apache.lucene.store.AlreadyClosedException exceptions in lucene/neo4j with "this IndexReader is closed"
# were no longer seen. Note that "legacy" explicit lucene indexes are used.
# see https://github.com/globalbioticinteractions/globalbioticinteractions/issues/701
# Suspect that lucene or other parts is not happy with read_only restrictions
# imposed by a read-only dataset.
#
dbms.read_only=false
...
```

Please advise on how to provide a public, read-only, neo4j instance without having to introduce an authentication wall.

**Comment by Roxling** [comment]
@jhpoelen Hello!
3.4 has reached its end-of-life and we are no longer patching/maintaining that branch. We have done many improvements over the years and it is very likely it has already been fixed in a later version. If you can reproduce this issue on a later version we actively maintain (preferably the latest), we will investigate it. You can find a list of supported (actively maintained) versions here https://neo4j.com/developer/kb/neo4j-supported-versions/

**Comment by jhpoelen** [comment]
@Roxling thanks for taking the time to respond. I understand that maintaining older versions is quite laborious, and you'll have to draw a line at some point.

Can you please mark this issue as "do not fix"? For some reason, I was unable to assign labels to this issue.

Closing issue.

**Comment by jhpoelen** [comment]
Note that, after upgrading to neo4j v3.5.29, the issue "this IndexReader is closed" has not appeared yet. This suggests that the issue has been fixed in some neo4j version after v3.4.18 .
